diff --git a/client/client.go b/client/client.go
index db4325e..09a5061 100644
--- a/client/client.go
+++ b/client/client.go
@@@ ============================================================================
@@@ This patch enables Kubesnap Docker Collector to work on CentOS 7
@@@ instead of GCE.
@@@ ============================================================================
@@ -186,7 +186,7 @@ func isHost(id string) bool {
 }
 
 func GetSubsystemPath(subsystem string, id string) (string, error) {
-	slice := "system.slice"
+	slice := "docker"
 	groupPath, err := cgroups.FindCgroupMountpoint(subsystem)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "[WARNING] Could not find mount point for %s\n", subsystem)
@@ -197,7 +197,7 @@ func GetSubsystemPath(subsystem string, id string) (string, error) {
 		groupPath = filepath.Join(groupPath, slice)
 
 		if !isHost(id) {
-			groupPath = filepath.Join(groupPath, "docker-"+id+".scope")
+			groupPath = filepath.Join(groupPath, id)
 		}
 		return groupPath, nil
 	}
