diff --git a/data/cifar10/get_cifar10.sh b/data/cifar10/get_cifar10.sh
index 623c848..515c446 100755
--- a/data/cifar10/get_cifar10.sh
+++ b/data/cifar10/get_cifar10.sh
@@ -6,7 +6,15 @@ cd $DIR
 
 echo "Downloading..."

-wget --no-check-certificate http://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz
+if [ -d /cache ]; then
+    if [ ! -f /cache/cifar-10-binary.tar.gz ]; then
+        wget -q --no-check-certificate http://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz -O /cache/cifar-10-binary.tar.gz
+    fi
+elif
+    wget -q --no-check-certificate http://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz -O ./cifar-10-binary.tar.gz
+fi
+
+cp /cache/cifar-10-binary.tar.gz .
 
 echo "Unzipping..."
