diff --git a/data/cifar10/get_cifar10.sh b/data/cifar10/get_cifar10.sh
index 623c848..698e9a9 100755
--- a/data/cifar10/get_cifar10.sh
+++ b/data/cifar10/get_cifar10.sh
@@ -5,8 +5,10 @@ DIR="$( cd "$(dirname "$0")" ; pwd -P )"
 cd $DIR
 
 echo "Downloading..."
-
-wget --no-check-certificate http://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz
+if [ ! -f /cache/cifar-10-binary.tar.gz ]; then
+    wget -q --no-check-certificate http://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz -O /cache/cifar-10-binary.tar.gz
+fi
+cp /cache/cifar-10-binary.tar.gz .
 
 echo "Unzipping..."

